webapp/components/_new/generic/BaseCard/BaseCard.vue

```diff
@@ -0,0 +1,121 @@
+<template>
+  <article :class="classNames">
+    <template v-if="$slots.imageColumn">
+      <aside class="image-column">
+        <slot name="imageColumn" />
+      </aside>
+      <section class="content-column">
+        <slot />
+      </section>
+    </template>
+
+    <template v-else-if="$slots.heroImage">
+      <section class="hero-image">
+        <slot name="heroImage" />
+      </section>
+      <slot />
+    </template>
+
+    <slot v-else />
+    <aside v-if="$slots.topMenu" class="top-menu">
+      <slot name="topMenu" />
+    </aside>
+  </article>
+</template>
+
+<script>
+export default {
+  props: {
+    highlight: {
+      type: Boolean,
+      default: false,
+    },
+    wideContent: {
+      type: Boolean,
+      default: false,
+    },
+  },
+  computed: {
+    classNames() {
+      let classNames = 'base-card'
+
+      if (this.$slots.imageColumn) classNames += ' --columns'
+      if (this.highlight) classNames += ' --highlight'
+      if (this.wideContent) classNames += ' --wide-content'
+
+      return classNames
+    },
+  },
+}
+</script>
+
+<style lang="scss">
+.base-card {
+  position: relative;
+  padding: $space-base;
+  border-radius: $border-radius-x-large;
+  overflow: hidden;
+  background-color: $color-neutral-100;
+  box-shadow: $box-shadow-base;
+
+  &.--columns {
+    display: flex;
+  }
+
+  &.--highlight {
+    border: $border-size-base solid $color-warning;
+  }
+
+  &.--wide-content {
+    padding: $space-small;
+
+    > .hero-image {
+      width: calc(100% + (2 * #{$space-small}));
+      margin: -$space-small;
```

Comment:
this is confusing for me... could you explain? you are setting the width to above 100% then adding negative margin to counteract that, then finally apply the margin-bottom that you want? :thinking:

Reply:
You almost got it right! :wink: I wouldn't usually do this or ask anyone to write CSS like this – but since we want to make it easy for people to use the `BaseCard` component, with or without a hero image, I decided to go with this solution as it requires no additional CSS or calculations when using the component.

It sets a `padding` on the `BaseCard` – but since we don't want a white frame around the hero image I need to counteract it with a negative `margin`. So far, so good, but parent `padding` is respected when setting a `width` of `100%`, so I need to add the padding value on top to make sure the image spans the whole width of the container. And then, as you said, I add a positive `margin-bottom` to make sure there's some space between the image and the card title.

Hope that makes sense now? It's not straightforward and I have thought about other solutions (e.g. not setting a `padding` on the `BaseCard` and instead adding `margin-left` and `margin-right` on all its children) but that got messy, too. So I thought it's better to have the complexity inside the component than to have to deal with it each time we want to use it.